app/routes/event.$slug.tsx

```diff
@@ -0,0 +1,31 @@
+import { LoaderFunctionArgs } from "@remix-run/node";
+import { json, useLoaderData } from "@remix-run/react";
+
+import { prisma } from "~/utils/db.server";
+
+export const loader = async ({ params }: LoaderFunctionArgs) => {
+  if (!params.slug) {
+    throw new Error("Missing eventid param");
+  }
+  const event = await prisma.events.findFirst({
+    where: {
+      slug: params.slug,
+    },
+  });
+  if (!event) {
+    throw new Response("Not Found", { status: 404 });
```

Comment:
Can follow the official docs: https://remix.run/docs/en/main/guides/not-found#how-to-send-a-404